src/ralph/deployment/deployment.py

```diff
@@ -397,23 +398,41 @@ def _assign_hostname(instance, new_hostname, net_env=None):
             _assign_hostname(instance, new_hostname, net_env)
 
 
-def check_ipaddress_unique(instances, data):
-    """
-    Validate if custom (other) IP is already assigned to another (base) object.
-    """
+def validate_ip_address(instances, data):
     if data['ip_or_network']['value'] == OTHER:
         assert len(instances) == 1
         instance = instances[0]
         address = data['ip_or_network'][OTHER]
-        try:
-            ip = IPAddress.objects.get(address=address)
-        except IPAddress.DoesNotExist:
-            pass
-        else:
-            if ip.ethernet and ip.ethernet.base_object_id != instance.pk:
-                raise ValidationError(
-                    'IP {} is already assigned to other object!'.format(address)
-                )
+        check_ip_from_defined_network(address)
+        check_ipaddress_unique(instance, address)
+
+
+def check_ipaddress_unique(instance, address):
+    """
+    Validate if `address` is already assigned to another (base) object.
+    """
+    try:
+        ip = IPAddress.objects.get(address=address)
+    except IPAddress.DoesNotExist:
+        pass
+    else:
+        if ip.ethernet and ip.ethernet.base_object_id != instance.pk:
+            raise ValidationError(
+                'IP {} is already assigned to other object!'.format(address)
+            )
+
+
+def check_ip_from_defined_network(address):
+    """
+    Validate if `address` belongs to any network already added
+    """
+    ip = ipaddress.ip_address(address)
+    try:
+        Network.objects.get(min_ip__lte=int(ip), max_ip__gte=int(ip))
```

Comment:
there could be many networks to which this ip will fit - we should use filter here